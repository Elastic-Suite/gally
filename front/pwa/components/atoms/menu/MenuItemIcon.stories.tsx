import { ComponentMeta, ComponentStory } from '@storybook/react'

import MenuItemIconComponent from './MenuItemIcon'

export default {
  title: 'Atoms/Menu',
  component: MenuItemIconComponent,
  argTypes: {
    childPadding: {
      description: 'Do not render a link',
    },
    code: {
      description: 'Also used as the icon name',
    },
    isActive: {
      description: 'Only works if childPadding is false',
    },
    isRoot: {
      description:
        'Only works if childPadding and sidebarStateTimeout are true',
    },
  },
} as ComponentMeta<typeof MenuItemIconComponent>

const Template: ComponentStory<typeof MenuItemIconComponent> = (args) => (
  <MenuItemIconComponent {...args} />
)

export const MenuItemIcon = Template.bind({})
MenuItemIcon.args = {
  childPadding: false,
  code: 'airplane-outline',
  href: '#',
  isActive: false,
  isRoot: false,
  label: 'Label',
  lightStyle: false,
  sidebarState: false,
  sidebarStateTimeout: false,
}
