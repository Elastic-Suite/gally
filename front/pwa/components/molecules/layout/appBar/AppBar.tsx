import { styled } from '@mui/system'
import { IMenu } from 'shared'

import Breadcrumbs from '~/components/atoms/breadcrumb/Breadcrumbs'
import User from '~/components/molecules/layout/user/User'

const CustomRoot = styled('div')(({ theme }) => ({
  display: 'flex',
  flexDirection: 'row',
  justifyContent: 'space-between',
  alignItems: 'center',
  paddingTop: theme.spacing(1.25),
  paddingBottom: theme.spacing(1.25),
  paddingRight: theme.spacing(4),
  paddingLeft: theme.spacing(4),
  position: 'fixed',
  right: 0,
  zIndex: '9',
  height: theme.spacing(8),
  backgroundColor: theme.palette.background.page,
  left: 'inherit',
  '&::before': {
    content: '""',
    position: 'absolute',
    border: '1px solid',
    borderColor: theme.palette.colors.neutral['300'],
    bottom: 0,
    width: `calc(100% - 64px)`,
  },
}))

interface IProps {
  slug?: string | string[]
  menu?: IMenu
  isConnectedWithValidJwt: boolean
}

function AppBar({ slug, menu, isConnectedWithValidJwt }: IProps): JSX.Element {
  return (
    <CustomRoot>
      <Breadcrumbs slug={slug} menu={menu} />
      <User isConnectedWithValidJwt={isConnectedWithValidJwt} />
    </CustomRoot>
  )
}

export default AppBar
