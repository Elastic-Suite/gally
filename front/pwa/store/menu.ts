import { PayloadAction, createSlice } from '@reduxjs/toolkit'

import { IFetch, IMenu, LoadStatus } from 'gally-admin-shared'

import { RootState } from './store'

export interface IMenuState {
  menu: IFetch<IMenu>
  menuItemActive: string
  sidebarState: boolean
  sidebarStateTimeout: boolean
  childrenState: Record<string, boolean>
}

const initialState: IMenuState = {
  menu: {
    status: LoadStatus.IDLE,
    data: {
      hierarchy: [],
    },
  },
  menuItemActive: '',
  sidebarState: true,
  sidebarStateTimeout: false,
  childrenState: {},
}

const menuSlice = createSlice({
  name: 'menu',
  initialState,
  reducers: {
    setChildState(
      state,
      action: PayloadAction<{ code: string; value: boolean }>
    ) {
      state.childrenState[action.payload.code] = action.payload.value
    },
    setMenu(state, action: PayloadAction<IFetch<IMenu>>) {
      state.menu = {
        ...state.menu,
        ...action.payload,
      }
    },
    setMenuItemActive(state, action: PayloadAction<string>) {
      state.menuItemActive = action.payload
    },
    setSidebarState(state, action: PayloadAction<boolean>) {
      state.sidebarState = action.payload
    },
    setSidebarStateTimeout(state, action: PayloadAction<boolean>) {
      state.sidebarStateTimeout = action.payload
    },
  },
})

export const {
  setChildState,
  setMenu,
  setMenuItemActive,
  setSidebarState,
  setSidebarStateTimeout,
} = menuSlice.actions
export const menuReducer = menuSlice.reducer

export const selectChildrenState = (
  state: RootState
): Record<string, boolean> => state.menu.childrenState
export const selectMenu = (state: RootState): IMenu => state.menu.menu.data
export const selectMenuItemActive = (state: RootState): string =>
  state.menu.menuItemActive
export const selectSidebarState = (state: RootState): boolean =>
  state.menu.sidebarState
export const selectSidebarStateTimeout = (state: RootState): boolean =>
  state.menu.sidebarStateTimeout
