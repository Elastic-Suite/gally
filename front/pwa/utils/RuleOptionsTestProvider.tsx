import { ReactNode } from 'react'

import ruleEngineOperator from '~/public/mocks/rule_engine_operators.json'
import sourceFields from '~/public/mocks/source_fields.json'
import sourceFieldLabels from '~/public/mocks/source_field_labels.json'
import { RuleAttributeType } from 'shared'

import RuleOptionsProvider from '~/components/stateful-providers/RuleOptionsProvider/RuleOptionsProvider'

interface IProps {
  children: ReactNode
}

const sourceFieldLabelsMap = new Map(
  sourceFieldLabels['hydra:member'].map((label) => [
    label.sourceField,
    label.label,
  ])
)

const fields = sourceFields['hydra:member'].map((field) => {
  const label = sourceFieldLabelsMap.get(`/source_fields/${field.id}`)
  return {
    id: field.id,
    code: field.code,
    label: (label || field.defaultLabel) ?? '',
    type: field.type as RuleAttributeType,
  }
})

function RuleOptionsTestProvider(props: IProps): JSX.Element {
  const { children } = props
  return (
    <RuleOptionsProvider
      catalogId={-1}
      localizedCatalogId={-1}
      fields={fields}
      ruleOperators={ruleEngineOperator}
    >
      {children}
    </RuleOptionsProvider>
  )
}

export default RuleOptionsTestProvider
