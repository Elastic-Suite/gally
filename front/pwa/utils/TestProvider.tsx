import { ReactNode } from 'react'

import { schemaContext, userContext } from '~/contexts'
import { IApi, IUser } from 'shared'

import OptionsProvider from '~/components/stateful-providers/OptionsProvider/OptionsProvider'

interface IProps {
  api: IApi
  children: ReactNode
}

const timestamp = Math.floor(Date.now() / 1000)
const expires = 8 * 60 * 60 // 8 hours

const user = {
  iat: timestamp,
  exp: timestamp + expires,
  roles: ['ROLE_ADMIN', 'ROLE_CONTRIBUTOR'],
  username: 'admin@example.com',
} as IUser

function TestProvider(props: IProps): JSX.Element {
  const { api, children } = props
  return (
    <userContext.Provider value={user}>
      <schemaContext.Provider value={api}>
        <OptionsProvider>{children}</OptionsProvider>
      </schemaContext.Provider>
    </userContext.Provider>
  )
}

export default TestProvider
