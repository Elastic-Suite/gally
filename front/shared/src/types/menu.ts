export interface IMenuChild {
  code: string
  label: string
  children?: IMenuChild[]
}

export interface IMenu {
  hierarchy: IMenuChild[]
}
