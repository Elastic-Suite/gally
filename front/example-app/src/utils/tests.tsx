import { RenderOptions, render, renderHook } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import { ReactNode } from 'react'
import { api } from 'gally-admin-shared'

import TestProvider from './TestProvider'

export type IExtendedRenderOptions = Omit<RenderOptions, 'queries'>

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderWithProviders(
  ui: React.ReactElement,
  renderOptions: IExtendedRenderOptions = {}
) {
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <MemoryRouter>
        <TestProvider api={api}>{children}</TestProvider>
      </MemoryRouter>
    )
  }
  return render(ui, { wrapper: Wrapper, ...renderOptions })
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderHookWithProviders<R>(
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  hook: (props: any) => R,
  renderOptions: IExtendedRenderOptions = {}
) {
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <MemoryRouter>
        <TestProvider api={api}>{children}</TestProvider>
      </MemoryRouter>
    )
  }
  return renderHook(hook, { wrapper: Wrapper, ...renderOptions })
}
