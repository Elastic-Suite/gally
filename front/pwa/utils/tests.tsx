import { RenderOptions, render, renderHook } from '@testing-library/react'
import { PreloadedState } from '@reduxjs/toolkit'
import { ReactNode } from 'react'
import { Bundle, IUser, api } from 'shared'

import { AppStore, RootState, setupStore } from '~/store'

import AppProvider from '~/components/stateful-providers/AppProvider/AppProvider'

import TestProvider from './TestProvider'

export interface IExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>
  store?: AppStore
}

export interface IRenderOutput {
  store: AppStore
}

const timestamp = Math.floor(Date.now() / 1000)
const expires = 8 * 60 * 60 // 8 hours

const user = {
  iat: timestamp,
  exp: timestamp + expires,
  roles: ['ROLE_ADMIN', 'ROLE_CONTRIBUTOR'],
  username: 'admin@example.com',
} as IUser

const defaultState = {
  data: {
    api,
    bundles: [Bundle.VIRTUAL_CATEGORY],
    configurations: ['https://localhost/products'],
  },
  user: {
    requestedPath: '',
    token:
      'eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJpYXQiOjE2NTk5NTM1NDQsImV4cCI6MTY1OTk4MjM0NCwicm9sZXMiOlsiUk9MRV9BRE1JTiIsIlJPTEVfQ09OVFJJQlVUT1IiXSwidXNlcm5hbWUiOiJhZG1pbkBleGFtcGxlLmNvbSJ9.ZNE2SHmtm9WgWEMmKGMjpcfEym2ysxC_CVcLtzlpuT3jlT0Zyz4VjoQHdTav9I7s9b2QzpeDNt3Sfrbka8IHcVb_TvZle_vII-TkQGal37RtXxKk_EMMQQlzLRO3zDWYy2I0ZPXv6iMlaSWOya4egOG2s1_tzFf4MnNNm2zEQbjzAtZ_Ij9UBMlSxK19AHvHlKU9AelA-NepjUdl6f4GxYPnyIwZw3PS294Wtffj7Kzw1ZZywDljl1xN_flg70cuO0GGkUqBNHrf7zBjPmXbAZZDAefoyQfVqB2v4GzDJfO847XymvBlHtCCjWx0ThK5afyOGL2MAAZkl-AoNjMcjA',
    user,
  },
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = defaultState,
    store,
    ...renderOptions
  }: IExtendedRenderOptions = {}
) {
  store = store ?? setupStore(preloadedState)
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <AppProvider store={store}>
        <TestProvider>{children}</TestProvider>
      </AppProvider>
    )
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderHookWithProviders<R>(
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  hook: (props: any) => R,
  {
    preloadedState = defaultState,
    store,
    ...renderOptions
  }: IExtendedRenderOptions = {}
) {
  store = store ?? setupStore(preloadedState)
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <AppProvider store={store}>
        <TestProvider>{children}</TestProvider>
      </AppProvider>
    )
  }
  return { store, ...renderHook(hook, { wrapper: Wrapper, ...renderOptions }) }
}
