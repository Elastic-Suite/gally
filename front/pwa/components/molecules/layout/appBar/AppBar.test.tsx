import { renderWithProviders } from '~/utils/tests'

import menu from '../../../../public/mocks/menu.json'

import AppBar from './AppBar'

describe('AppBar match snapshot', () => {
  it('AppBar simple', () => {
    const { container } = renderWithProviders(
      <AppBar slug={['search']} menu={menu} isConnectedWithValidJwt />
    )
    expect(container).toMatchSnapshot()
  })

  it('AppBarTwoSlug', () => {
    const { container } = renderWithProviders(
      <AppBar
        slug={['search', 'configuration']}
        menu={menu}
        isConnectedWithValidJwt
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('AppBarThreeSlug', () => {
    const { container } = renderWithProviders(
      <AppBar
        slug={['search', 'configuration', 'autocompletion']}
        menu={menu}
        isConnectedWithValidJwt
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('AppBarThreeSlug not connected', () => {
    const { container } = renderWithProviders(
      <AppBar
        slug={['search', 'configuration', 'autocompletion']}
        menu={menu}
        isConnectedWithValidJwt={false}
      />
    )
    expect(container).toMatchSnapshot()
  })
})
