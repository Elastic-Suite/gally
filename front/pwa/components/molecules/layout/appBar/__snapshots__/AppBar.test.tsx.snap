// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar match snapshot AppBar simple 1`] = `
<div>
  <div
    class="css-19ii7w2"
  >
    <nav
      aria-label="breadcrumb"
      class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-1sglq17-MuiTypography-root-MuiBreadcrumbs-root"
    >
      <ol
        class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
      >
        <li
          class="MuiBreadcrumbs-li"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-zhuy6w-MuiTypography-root"
          >
            <span
              class="css-6nl1o3"
            >
              Recherche
            </span>
          </p>
        </li>
      </ol>
    </nav>
    <div
      class="css-1gzl8uu"
    >
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="css-l3v3zn"
        >
          <ion-icon
            name="help-circle-outline"
            style="font-size: 18px; color: rgb(129, 135, 185);"
          />
        </div>
        <div
          class="MuiBox-root css-79elbk"
        >
          <div
            class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-199euq-MuiCollapse-root"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper MuiCollapse-vertical css-1642zpf-MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner MuiCollapse-vertical css-xxknsj-MuiCollapse-wrapperInner"
              >
                <div
                  class="css-c7ciki"
                >
                  <div
                    class="css-ykyq1f"
                  >
                    <div>
                      Helpdesk
                    </div>
                    <div>
                      User guide
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="css-1gbhtbv"
        >
          <ion-icon
            name="person-outline"
            style="font-size: 15px; color: rgb(129, 135, 185);"
          />
          <div
            class="css-1cn3y7o"
          >
            Admin Name
          </div>
          <div
            class="css-1tg3i43"
          >
            <ion-icon
              name="chevron-down"
              style="font-size: 15px; color: rgb(129, 135, 185);"
            />
          </div>
        </div>
        <div
          class="MuiBox-root css-79elbk"
        >
          <div
            class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-199euq-MuiCollapse-root"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper MuiCollapse-vertical css-1642zpf-MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner MuiCollapse-vertical css-xxknsj-MuiCollapse-wrapperInner"
              >
                <div
                  class="css-1vcaigq"
                >
                  <div
                    class="css-9inyfk"
                  >
                    <div
                      class="css-1bt56fy"
                    >
                      Admin name
                    </div>
                    <div
                      class="css-3lsmy"
                    >
                      adminame@mail.com
                    </div>
                    <div
                      class="css-1772pc6"
                    />
                    <div
                      class="css-12w2p0s"
                    >
                      Account
                    </div>
                    <div
                      class="css-12w2p0s"
                    >
                      Log out
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`AppBar match snapshot AppBarThreeSlug 1`] = `
<div>
  <div
    class="css-19ii7w2"
  >
    <nav
      aria-label="breadcrumb"
      class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-1sglq17-MuiTypography-root-MuiBreadcrumbs-root"
    >
      <ol
        class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
      >
        <li
          class="MuiBreadcrumbs-li"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-zhuy6w-MuiTypography-root"
          >
            <span
              class="css-1epdhd3"
            >
              Recherche
            </span>
          </p>
        </li>
        <li
          aria-hidden="true"
          class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
        >
          /
        </li>
        <li
          class="MuiBreadcrumbs-li"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-zhuy6w-MuiTypography-root"
          >
            <span
              class="css-1epdhd3"
            >
              Configuration
            </span>
          </p>
        </li>
        <li
          aria-hidden="true"
          class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
        >
          /
        </li>
        <li
          class="MuiBreadcrumbs-li"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-zhuy6w-MuiTypography-root"
          >
            <span
              class="css-6nl1o3"
            >
              Autocompl√©tion
            </span>
          </p>
        </li>
      </ol>
    </nav>
    <div
      class="css-1gzl8uu"
    >
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="css-l3v3zn"
        >
          <ion-icon
            name="help-circle-outline"
            style="font-size: 18px; color: rgb(129, 135, 185);"
          />
        </div>
        <div
          class="MuiBox-root css-79elbk"
        >
          <div
            class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-199euq-MuiCollapse-root"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper MuiCollapse-vertical css-1642zpf-MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner MuiCollapse-vertical css-xxknsj-MuiCollapse-wrapperInner"
              >
                <div
                  class="css-c7ciki"
                >
                  <div
                    class="css-ykyq1f"
                  >
                    <div>
                      Helpdesk
                    </div>
                    <div>
                      User guide
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="css-1gbhtbv"
        >
          <ion-icon
            name="person-outline"
            style="font-size: 15px; color: rgb(129, 135, 185);"
          />
          <div
            class="css-1cn3y7o"
          >
            Admin Name
          </div>
          <div
            class="css-1tg3i43"
          >
            <ion-icon
              name="chevron-down"
              style="font-size: 15px; color: rgb(129, 135, 185);"
            />
          </div>
        </div>
        <div
          class="MuiBox-root css-79elbk"
        >
          <div
            class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-199euq-MuiCollapse-root"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper MuiCollapse-vertical css-1642zpf-MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner MuiCollapse-vertical css-xxknsj-MuiCollapse-wrapperInner"
              >
                <div
                  class="css-1vcaigq"
                >
                  <div
                    class="css-9inyfk"
                  >
                    <div
                      class="css-1bt56fy"
                    >
                      Admin name
                    </div>
                    <div
                      class="css-3lsmy"
                    >
                      adminame@mail.com
                    </div>
                    <div
                      class="css-1772pc6"
                    />
                    <div
                      class="css-12w2p0s"
                    >
                      Account
                    </div>
                    <div
                      class="css-12w2p0s"
                    >
                      Log out
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`AppBar match snapshot AppBarTwoSlug 1`] = `
<div>
  <div
    class="css-19ii7w2"
  >
    <nav
      aria-label="breadcrumb"
      class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-1sglq17-MuiTypography-root-MuiBreadcrumbs-root"
    >
      <ol
        class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
      >
        <li
          class="MuiBreadcrumbs-li"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-zhuy6w-MuiTypography-root"
          >
            <span
              class="css-1epdhd3"
            >
              Recherche
            </span>
          </p>
        </li>
        <li
          aria-hidden="true"
          class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
        >
          /
        </li>
        <li
          class="MuiBreadcrumbs-li"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-zhuy6w-MuiTypography-root"
          >
            <span
              class="css-6nl1o3"
            >
              Configuration
            </span>
          </p>
        </li>
      </ol>
    </nav>
    <div
      class="css-1gzl8uu"
    >
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="css-l3v3zn"
        >
          <ion-icon
            name="help-circle-outline"
            style="font-size: 18px; color: rgb(129, 135, 185);"
          />
        </div>
        <div
          class="MuiBox-root css-79elbk"
        >
          <div
            class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-199euq-MuiCollapse-root"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper MuiCollapse-vertical css-1642zpf-MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner MuiCollapse-vertical css-xxknsj-MuiCollapse-wrapperInner"
              >
                <div
                  class="css-c7ciki"
                >
                  <div
                    class="css-ykyq1f"
                  >
                    <div>
                      Helpdesk
                    </div>
                    <div>
                      User guide
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="css-1gbhtbv"
        >
          <ion-icon
            name="person-outline"
            style="font-size: 15px; color: rgb(129, 135, 185);"
          />
          <div
            class="css-1cn3y7o"
          >
            Admin Name
          </div>
          <div
            class="css-1tg3i43"
          >
            <ion-icon
              name="chevron-down"
              style="font-size: 15px; color: rgb(129, 135, 185);"
            />
          </div>
        </div>
        <div
          class="MuiBox-root css-79elbk"
        >
          <div
            class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-199euq-MuiCollapse-root"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper MuiCollapse-vertical css-1642zpf-MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner MuiCollapse-vertical css-xxknsj-MuiCollapse-wrapperInner"
              >
                <div
                  class="css-1vcaigq"
                >
                  <div
                    class="css-9inyfk"
                  >
                    <div
                      class="css-1bt56fy"
                    >
                      Admin name
                    </div>
                    <div
                      class="css-3lsmy"
                    >
                      adminame@mail.com
                    </div>
                    <div
                      class="css-1772pc6"
                    />
                    <div
                      class="css-12w2p0s"
                    >
                      Account
                    </div>
                    <div
                      class="css-12w2p0s"
                    >
                      Log out
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
